renderdoc/driver/vulkan/vk_initstate.cpp

```diff
@@ -933,7 +933,20 @@ bool WrappedVulkan::Serialise_InitialState(SerialiserType &ser, ResourceId id, W
 
     // unmap the resource we mapped before - we need to do this on read and on write.
     if(!IsStructuredExporting(m_State) && mappedMem.mem != VK_NULL_HANDLE)
+    {
+      VkMappedMemoryRange range = {
+          VK_STRUCTURE_TYPE_MAPPED_MEMORY_RANGE,
+          NULL,
+          Unwrap(mappedMem.mem),
+          mappedMem.offs,
+          mappedMem.size,
+      };
+
+      vkr = ObjDisp(d)->FlushMappedMemoryRanges(Unwrap(d), 1, &range);
```

Comment:
There's a slight subtlety with this - the code is written to be serialising bidirectionally. So during capture, this is reading directly from mapped memory and going through the compression to disk. During replay, this is reading out of the file decompression and writing directly into the mapped memory.

So the flush here is fine as long as it's guarded by a `if(IsReplayingAndReading())` for the disk-read & GPU-write case, but I think we'll also need a corresponding read e.g. up in the `if(ser.IsWriting())` check above (line 900) where we do GPU-read & disk-write.

This also applies in the same way to the case below in `vk_sparse_initstate.cpp`.

Reply:
Ok, thanks. I don't have time to address that today, so will look again tomorrow sometime.

Reply:
No problem! Thanks for looking at this.